CONTRIBUTING.md

```diff
@@ -0,0 +1,181 @@
+# Contributing to Chaos Mesh
+
+Thanks for your interest in improving the project! This document provides a step-by-step guide for general contributions to Chaos Mesh.
+
+## Communications
+
+Before starting work on something major, please reach out to us via GitHub, Slack, email, etc. We will make sure no one else is already working on it and ask you to open a GitHub issue. Also, we will provide necessary guidance should you need it.
+
+Specifically, if you want to develop a specific chaos type, åyou may also find [Development Guide](https://chaos-mesh.org/docs/development_guides/development_overview) useful.
+
+## Submitting a PR
+
+If you have a specific idea of a fix or update, follow these steps below to submit a PR:
+
+- [Step 1: Make the change](#step-1-make-the-change)
+- [Step 2: Run unit tests](#step-2-run-unit-tests)
+- [Step 3: Run e2e test](#step-3-run-e2e-test)
+- [Step 4: Start Chaos Mesh locally and do manual tests](#step-4-start-chaos-mesh-locally-and-do-manual-tests)
+- [Step 5: Commit and push your changes](#step-5-commit-and-push-your-changes)
+- [Step 6: Create a pull request](#step-6-create-a-pull-request)
+- [Step 7: Get a code review](#step-7-get-a-code-review)
+
+### Step 1: Make the change
+
+1. Fork the Chaos Mesh repo, and then clone it to your local:
+
+```bash
+$ export user={your github. profile name}
+$ git clone https://github.com/${user}/chaos-mesh.git
+```
+
+1. Set your cloned local to track the upstream repository:
+
+```bash
+$ cd chaos-mesh
+$ git remote add upstream https://github.com/pingcap/chaos-mesh
+```
+
+3. Get your local master up-to-date and create your working branch:
+
+```bash
+$ git fetch upstream
+$ git checkout master
+$ git rebase upstream/master
+$ git checkout -b myfeature
+```
+
+4. Make the change on the code
+
+You can new edit the code on the `myfeature` branch.
+
+If you want to update the `crd.yam` according the the CRD structs, run the following commands:
+
+```bash
+$ make generate
+$ make yaml
+```
+
+5. Check the code change by running the following command:
+
+```bash
+$ make check
+```
+
+This will show errors if your code change does not pass the check. (eg: fmt, lint). Please fix them before submitting the PR.
+
+### Step 2: Run unit tests
+
+Before running your code in a real Kubernetes cluster, make sure it passes all unit tests:
+
+```bash
+$ make ensure-kubebuilder # install some test dependencies
+$ make test
+```
+
+### Step 3: Run e2e test
+
+Chaos Mesh code must pass e2e test before being submitted. Before started, you must have [Docker](https://www.docker.com/get-started/) installed and running.
+
```

Comment:
I am wondering whether we really need to add this step. Running e2e locally is not easy and it is really slow...

For users who want to do contributions, open a pr and run e2e test via Jenkins is much more convenient

Reply:
> I am wondering whether we really need to add this step. Running e2e locally is not easy and it is really slow...
> 
> For users who want to do contributions, open a pr and run e2e test via Jenkins is much more convenient

Agree. I always open a pr and run e2e test via Jenkins.